### PR TITLE
add manual_ok_or lint

### PR DESCRIPTION
**Pull request by montrivo**
Implements partially #5923

changelog: add lint manual_ok_or

### CODE REVIEW

**Review by flip1995 — COMMENTED** [review]
Couldn't you implement this lint in the `methods` module? You wouldn't need the whole method call extraction boilerplate there.

**Comment by montrivo** [comment]
> Couldn't you implement this lint in the `methods` module? You wouldn't need the whole method call extraction boilerplate there.

In a next iteration, I planned to implement the pattern matching case. See comment about this in the test file.
```rust
intOpt match {
  Some(v) => Ok(v),
  None => Err("error"),
}
```
This doesn't really fit in the `methods` module, is it?

**Comment by flip1995** [comment]
Ah ok, I missed that. Then having this in a separate file is fine.

File: clippy_lints/src/manual_ok_or.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            if is_ok_wrapping(cx, &args[2]);
+            if let ExprKind::Call(Expr { kind: ExprKind::Path(err_path), .. }, &[ref err_arg]) = or_expr.kind;
+            if utils::match_qpath(err_path, &utils::paths::RESULT_ERR);
+            if let Some(method_receiver_snippet) = utils::snippet_opt(cx, method_receiver.span);
+            if let Some(err_arg_snippet) = utils::snippet_opt(cx, err_arg.span);
+            if let Some(indent) = utils::indent_of(cx, scrutinee.span);
+            then {
+                let reindented_err_arg_snippet =
+                    utils::reindent_multiline(err_arg_snippet.into(), true, Some(indent + 4));
+                utils::span_lint_and_sugg(
```
Comment (flip1995) [inline clippy_lints/src/manual_ok_or.rs:63]: NIT: please import this instead. Same for the other utils functions.
```suggestion
                span_lint_and_sugg(
```

**Review by flip1995 — APPROVED** [review]

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**